tests/robustness/failpoint/cluster.go

```diff
@@ -138,6 +145,80 @@ func (f memberReplace) Available(config e2e.EtcdProcessClusterConfig, member e2e
 	return config.ClusterSize > 1 && (config.Version == e2e.QuorumLastVersion || member.Config().ExecPath == e2e.BinPath.Etcd)
 }
 
+type memberDowngrade struct{}
+
+func (f memberDowngrade) Inject(ctx context.Context, t *testing.T, lg *zap.Logger, clus *e2e.EtcdProcessCluster, baseTime time.Time, ids identity.Provider) ([]report.ClientReport, error) {
+	v, err := e2e.GetVersionFromBinary(e2e.BinPath.Etcd)
+	if err != nil {
+		return nil, err
+	}
+	targetVersion := semver.Version{Major: v.Major, Minor: v.Minor - 1}
+	numberOfMembersToDowngrade := rand.Int()%len(clus.Procs) + 1
+	membersToDowngrade := rand.Perm(len(clus.Procs))[:numberOfMembersToDowngrade]
+	lg.Info("Test downgrading members", zap.Any("members", membersToDowngrade))
+
+	member := clus.Procs[0]
+	endpoints := []string{member.EndpointsGRPC()[0]}
+	cc, err := clientv3.New(clientv3.Config{
+		Endpoints:            endpoints,
+		Logger:               zap.NewNop(),
+		DialKeepAliveTime:    10 * time.Second,
+		DialKeepAliveTimeout: 100 * time.Millisecond,
+	})
+	if err != nil {
+		return nil, err
+	}
+	defer cc.Close()
+
+	// Need to wait health interval for cluster to accept changes
+	time.Sleep(etcdserver.HealthInterval)
+	lg.Info("Enable downgrade")
+	err = enableDowngrade(ctx, cc, &targetVersion)
+	if err != nil {
+		return nil, err
+	}
+	// Need to wait health interval for cluster to prepare for downgrade
+	time.Sleep(etcdserver.HealthInterval)
+
+	for _, memberID := range membersToDowngrade {
+		member = clus.Procs[memberID]
+		lg.Info("Downgrading member", zap.String("member", member.Config().Name))
+		if err = member.Stop(); err != nil {
+			return nil, err
+		}
+		member.Config().ExecPath = e2e.BinPath.EtcdLastRelease
+		err = patchArgs(member.Config().Args, "initial-cluster-state", "existing")
```

Comment:
It isn't necessary to patch the flag `--initial-cluster-state`, because the data-dir and wal-dir have already been created & initialised. It won't do any harm to do it, but suggest to remove it for simplicity.